02_knight_and_princess/tamara_gadieva/main.cpp

```diff
@@ -0,0 +1,32 @@
+#include "graph_generate.hpp"
+
+int main() {
+  const int vertices_num = 14;
+  const int edges_num = 18;
+
+  //информация о парах вершин, соединённых рёбрами
+  const std::vector<std::pair<int, int>> vertex_pairs = {
+      {0, 1},  {0, 2},  {0, 3},  {1, 4},   {1, 5},   {1, 6},
+      {2, 7},  {2, 8},  {3, 9},  {4, 10},  {5, 10},  {6, 10},
+      {7, 11}, {8, 11}, {9, 12}, {10, 13}, {11, 13}, {12, 13}};
+
+  Graph graph;
+
+  for (int i = 0; i < vertices_num; i++) {
+    graph.add_vertex();
+  }
+
+  for (auto cur_pair : vertex_pairs) {
```

Comment:
1. Что такое `cur`? `current`? 
2. `const` ссылка
3. pair используется со множественным числом: "пара ботинок", "пара вершин". Соответственно `vertices` а не `vertex`
```suggestion
  for (const auto& vertices_pair : vertices_pairs) {
```

Когда ты проходишься по элементам, не нужно особо выдумывать название, обычно это присто единственное число от названия коллекции:
- `vertex : vertices`
- `edge : edges`
- `pair : paris`